lib/rspec/mocks/message_expectation.rb

```diff
@@ -85,6 +85,48 @@ def and_return(first_value, *values)
         nil
       end
 
+      # Tells the object to invoke a Proc when it receives the message. Given
+      # more than one value, the result of the first Proc is returned the first
+      # time the message is received, the result of the second Proc is returned
+      # the next time, etc, etc.
+      #
+      # If the message is received more times than there are Procs, the result of
+      # the last Proc is returned for every subsequent call.
+      #
```

Comment:
We seem to pass message args to that proc, does it make sense to reflect this in the docs?